content/recipes/authoring/kata-snippets/full-tests.md

```diff
@@ -7,4 +7,57 @@ next: /recipes/authoring/kata-snippets/preloaded/
 
 # Writing Full Test Suite
 
-_This page is a stub._
\ No newline at end of file
+This article contains a set of guidelines, which can be used by kata authors to create good test suites for their kata. They were collected to help to ensure that kata, translations, and their test suites are of sufficient quality and users' experience with tests will be as good as possible. Failing to comply with below guidelines should be reported as an issue with request for fix, preferrably during a review before the content gets approved. In case of severe violations, affected kata or translation can be retired, moved back to beta state, or rejected.
+
+NOTE: There are many kinds of kata, and some guidelines might simply not apply to some of them. There are puzzles, hackmes, kata which mess with internals of language runtime, and some types of challenges require special ways of testing. But still, guidelines found below apply to majority of kata in some extent, and when used, kata author can be sure that their tests will be clear, efficient, reliable, and easy to debug.  
+
+## General Guidelines
+
+- Tests are code, and as such, they should **conform to [General Coding Guidelines](/recipes/authoring/kata-snippets/coding-general/)**.
+- **Be familiar with the testing framework used by Codewars for your language.** Some languages use traditional, widely known testing frameworks, like `JUnit` for Java or `chai` for JavaScript. Some use frameworks dedicated specifically for use by Codewars (for example Python). Know what assertions are there, how to test objects, types, collections, etc. Know techniques provided by the framework (test case generators, parametrized test cases, etc.) and assertion libraries. You can visit the [reference page for your language](/languages/) to find more detailed information.
+- **Test suite should be organized** by utilizing API of your testing framework. Frameworks use `Describe` blocks, test fixtures, categories, or other means, to collect test cases into manageable blocks, label them, etc.
```

Comment:
Yes, I am trying to stay out of specifics (it's `Describe` for C test suites ;) ) because approaches differ between test frameworks.

C, Python, JS use granularity levels of describe/it/assert, for Java and C# it's test fixture/test method, for other languages it can be labelled categories or something else, and the levels do not always correspond between approaches. I want to stay away from this in my description and all I want to tell is "organize your tests". Specific hints should be possible to find in docs related to specific language or test framework reference.

Reply:
ok :+1: (I was too much python/JS oriented, there)